autoload/youcompleteme/hierarchy.vim

```diff
@@ -135,25 +140,76 @@ function! s:SetUpMenu()
     \   maxheight: &lines * 75/100,
     \   scrollbar: 1,
     \   padding: [ 0, 0, 0, 0 ],
-    \ } )
+    \   highlight: 'Normal',
+    \   border: [],
+    \ }
+  if &ambiwidth ==# 'single' && &encoding ==? 'utf-8'
+    let opts[ 'borderchars' ] = [ '─', '│', '─', '│', '╭', '╮', '╯', '╰' ]
+  endif
+
+  let s:popup_id = popup_create( [], opts )
```

Comment:
We could postpone calling `popup_create()` until we have all the lines, then call `popup_create( lines, opts )`.

Not sure if that could ever be noticable from the user's perspective.

Reply:
We create it first so that we can get the ‘core_width’ which is the width minus padding and scrollbar etc.